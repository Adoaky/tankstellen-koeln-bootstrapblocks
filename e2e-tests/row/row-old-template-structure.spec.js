/**
 * WordPress dependencies
 */
import {
	activatePlugin,
	deactivatePlugin,
	createNewPost,
	searchForBlock,
	getAllBlocks,
	selectBlockByClientId,
	getEditedPostContent,
} from '@wordpress/e2e-test-utils'

describe( 'row block old template structure', () => {
	beforeAll( async () => {
		await activatePlugin( 'wp-bootstrap-blocks-test-row-old-template-structure' );
	} );

	afterAll( async () => {
		await deactivatePlugin( 'wp-bootstrap-blocks-test-row-old-template-structure' );
	} );

	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'Should warn if old template structure is used', async () => {
		const expectedMsg = 'wp-bootstrap-blocks: The old object template structure (<= v1.2.0) of the row block is deprecated, please migrate your templates to the new array structure (v1.3.0+). As soon as you have updated your template structure you need to disable the old object template structure with the wpBootstrapBlocks.row.useOldObjectTemplateStructure filter.';
		expect( console ).toHaveWarnedWith( expectedMsg );
	} );

	it( 'Custom template defined with old object structure should be available', async () => {
		// Insert row block
		await searchForBlock( 'Bootstrap Row' )
		await page.click( 'button.editor-block-list-item-wp-bootstrap-blocks-row' )

		expect( await getEditedPostContent() ).toMatchSnapshot();

		// Select row block
		await selectBlockByClientId(
			( await getAllBlocks() )[ 0 ].clientId
		);

		// Custom template should be available
		expect( await page.$$( '.wp-bootstrap-blocks-template-selector-button' ) ).toHaveLength(6);
		expect( await page.$( '.wp-bootstrap-blocks-template-selector-button > button[aria-label="1 Column (1/3 width)"]' ) ).not.toBeNull();

		// Template should be applied
		await page.click( '.wp-bootstrap-blocks-template-selector-button > button[aria-label="1 Column (1/3 width)"]' );

		expect( await getEditedPostContent() ).toMatchSnapshot();

		expect( console ).toHaveWarned();
	} );
} );
