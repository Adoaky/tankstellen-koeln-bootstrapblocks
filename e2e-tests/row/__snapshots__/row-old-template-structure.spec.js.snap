// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`row block old template structure Custom template defined with old object structure should be available 1`] = `
"<!-- wp:wp-bootstrap-blocks/row -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;

exports[`row block old template structure Custom template defined with old object structure should be available 2`] = `
"<!-- wp:wp-bootstrap-blocks/row {\\"template\\":\\"1-66percent\\"} -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":8} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;
